src/renderer/components/+catalog/catalog.tsx

```diff
@@ -91,16 +93,24 @@ class NonInjectedCatalog extends React.Component<Dependencies> {
   }
 
   async componentDidMount() {
+    const {
+      catalogEntityStore,
+      catalogPreviousActiveTabStorage,
+      catalogCategoryRegistry,
+    } = this.props;
+
     disposeOnUnmount(this, [
-      this.props.catalogEntityStore.watch(),
+      catalogEntityStore.watch(),
```

Comment:
Notice how unintuitive this UI-driven disposer code is to make `catalogEntityStore` do something. What if there are multiple "watchers" of `catalogEntityStore` in the application? What if one of them disposes its "watch"? That would involuntary end the watch for the first watch. We can do better.

Reply:
Yes and know, `catalogEntityStore` is really just state for this component. It isn't the same as the registry.